console/src/view/task.rs

```diff
@@ -123,13 +160,95 @@ impl TaskView {
         let mut fields = Text::default();
         fields.extend(task.formatted_fields().iter().cloned().map(Spans::from));
 
+        let chart_data = make_chart_data(self.details.clone(), task.id(), sparkline_area.width - 2);
+        let histogram_sparkline = Sparkline::default()
+            .block(Block::default().title("Poll Times").borders(Borders::ALL))
+            .data(&chart_data);
+
         let task_widget = Paragraph::new(metrics).block(block_for("Task"));
         let wakers_widget = Paragraph::new(vec![wakers, wakeups]).block(block_for("Waker"));
         let fields_widget = Paragraph::new(fields).block(block_for("Fields"));
 
+        let percentiles_widget = block_for("Poll Times Stats");
+        let (percentiles_1, percentiles_2) =
+            make_percentiles_widgets(self.details.clone(), task.id());
+        let percentiles_widget_1 = Paragraph::new(percentiles_1);
+        let percentiles_widget_2 = Paragraph::new(percentiles_2);
+
         frame.render_widget(Block::default().title(controls), controls_area);
         frame.render_widget(task_widget, stats_area[0]);
         frame.render_widget(wakers_widget, stats_area[1]);
         frame.render_widget(fields_widget, fields_area);
+        frame.render_widget(percentiles_widget, histogram_area[0]);
+        frame.render_widget(percentiles_widget_1, percentiles_columns[0]);
+        frame.render_widget(percentiles_widget_2, percentiles_columns[1]);
+        frame.render_widget(histogram_sparkline, sparkline_area);
+    }
+}
+
+fn make_chart_data(details: DetailsRef, task_id: u64, width: u16) -> Vec<u64> {
+    details
+        .borrow()
+        .as_ref()
+        .and_then(|details| filter_same_task(task_id, details))
+        .and_then(|details| details.poll_times_histogram())
+        .map(|histogram| {
+            // This is probably very buggy
+            let steps = ((histogram.max() - histogram.min()) as f64 / width as f64).ceil() as u64;
+            if steps > 0 {
+                let data: Vec<u64> = histogram
+                    .iter_linear(steps)
+                    .map(|it| it.count_since_last_iteration())
+                    .collect();
+                data
+            } else {
+                Vec::new()
+            }
+        })
+        .unwrap_or_default()
+}
+
+fn make_percentiles_widgets(details: DetailsRef, task_id: u64) -> (Text<'static>, Text<'static>) {
+    let percentiles_iter = details
+        .borrow()
+        .as_ref()
+        .and_then(|details| filter_same_task(task_id, details))
+        .and_then(|details| details.poll_times_histogram())
```

Comment:
nit, take it or leave it: this could be combined into one `and_then`
```suggestion
        .and_then(|details| Some(filter_same_task(task_id, details)?.poll_times_histogram()))
```
but, what you have now might be clearer, so it would be fine to leave it as-is...

Reply:
I tried this, but `histogram` in the next line in `.map(..)` is now an `Option<Histogram>` so I'd need to make more changes.

Reply:
oh i see, nvm!